crates/biome_service/src/file_handlers/astro.rs

```diff
@@ -0,0 +1,106 @@
+use crate::file_handlers::{
+    AnalyzerCapabilities, Capabilities, DebugCapabilities, ExtensionHandler, FormatterCapabilities,
+    Language, Mime, ParserCapabilities,
+};
+use crate::settings::SettingsHandle;
+use crate::WorkspaceError;
+use biome_formatter::Printed;
+use biome_fs::RomePath;
+use biome_js_formatter::format_node;
+use biome_js_parser::{parse_js_with_cache, JsParserOptions};
+use biome_js_syntax::{JsFileSource, JsLanguage};
+use biome_parser::AnyParse;
+use biome_rowan::{FileSource, NodeCache};
+use lazy_static::lazy_static;
+use regex::Regex;
+use tracing::{debug, error, info};
+
+#[derive(Debug, Default, PartialEq, Eq)]
+pub(crate) struct AstroFileHandler;
+
+lazy_static! {
+    pub static ref ASTRO_FENCE: Regex = Regex::new(r#"---\s*\n"#).unwrap();
```

Comment:
If we use multiline mode, [`^` and `$` matches start/end of line](https://docs.rs/regex/latest/regex/#empty-matches).

```suggestion
    pub static ref ASTRO_FENCE: Regex = Regex::new(r#"^---\s*$"#).unwrap();
```

I am not sure if it is enabled by default.
You certainly need to use `RegexBuilder::new`:

```rust
RegexBuilder::new(r#"^---\s*$"#).multi_line(true).build().unwrap()
```

Reply:
I applied you suggestions in [`4c3a31a` (#1718)](https://github.com/biomejs/biome/pull/1718/commits/4c3a31aafaa6d2cb13598ce7ff966e54e1ac65ed) , although I had to add `+ 1` because the newline wasn't being picked up by `start.end()`